ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -11484,14 +11484,13 @@ private Operator genTablePlan(String alias, QB qb) throws SemanticException {
       }
 
       // put all virtual columns in RowResolver.
-      if (!tab.isNonNative()) {
-        for (VirtualColumn vc : VirtualColumn.getRegistry(conf)) {
-          vcList.add(vc);
-          rwsch.put(alias, vc.getName().toLowerCase(), new ColumnInfo(vc.getName(),
-                  vc.getTypeInfo(), alias, true, vc.getIsHidden()
-          ));
-        }
+      vcList = VirtualColumn.getRegistry(conf);
```

Comment:
nit: The `vcList` has been declared previously. We are redefining it. Is it ok? Shall we just use another variable, to make sure no collision happens?